reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ToMap.kt

```diff
@@ -1,13 +1,22 @@
 package com.badoo.reaktive.observable
 
 import com.badoo.reaktive.single.Single
+import com.badoo.reaktive.utils.isFrozen
 
 fun <T, K> Observable<T>.toMap(keySelector: (T) -> K): Single<Map<K, T>> =
     toMap(keySelector) { it }
 
 fun <T, K, V> Observable<T>.toMap(keySelector: (T) -> K, valueSelector: (T) -> V): Single<Map<K, V>> =
-    collect(emptyMap()) { map, item ->
+    collect(LinkedHashMap()) { map, item ->
```

Comment:
Why `LinkedHashMap` and not just `HashMap`? RxJava uses `HashMap`.

Reply:
Because that's what Kotlin uses for `mapOf` and for copying `Map.plus`. If there are any benchmarks around occupied memory, we can check the real impact for optimization purposes. But you may want to check this out first: https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/

Reply:
Since the result is `Map` which is an interface there is no guarantee for iteration order. Actual map is implementation details. No one should assume anything about it. But lets use `LinkedHashMap` to be consistent in all cases.

Reply:
I'm not assuming anything, I use the default Kotlin implementation. And article was shared FYI to get an idea why LHM might be used in Kotlin